client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
 // limitations under the License.
 //
 
-= MicroProfile GraphQL Client
+= GraphQL Client for MicroProfile 
 
-Here add the client spec.
\ No newline at end of file
+== Rationale
+
+MicroProfile GraphQL 1.0 has been focused on the server-side enabling to develop and expose GraphQL endpoints. The purpose of this specification is to define a so-called "low level" client API. Low level means close to the GraphQL semantic, including for instance errors and partial results. We can compare with what JAX-RS client API is for REST.
+
+A "high level" client API, similar to MicroProfile RestClient, will be proposed later.
+
+== Goals
+
+1. **Provides full control over the MicroProfile GraphQL capabilities**: operations, arguments, variables, input types, scalars, interface, partial results, errors ...
+2. **HTTP support**: while the GraphQL specification is independent from the transport layer, we believe that in practice it is relevant to support HTTP features such as timeouts and headers
```

Comment:
Should we recommend to support POST first and see later whether accepting GET also make sense?

Reply:
I personally think it would actually make a lot of sense to use only GET for queries and only POST for mutations. But that's not how most GraphQL implementations work.